pkg/api/v1/atlascluster_types.go

```diff
@@ -30,6 +30,20 @@ func init() {
 	SchemeBuilder.Register(&AtlasCluster{}, &AtlasClusterList{})
 }
 
+type ProviderName string
+type ClusterType string
+
+const (
+	ProviderAWS    ProviderName = "AWS"
+	ProviderGCP    ProviderName = "GCP"
+	ProviderAzure  ProviderName = "AZURE"
+	ProviderTenant ProviderName = "TENANT"
+
+	TypeReplicaSet ClusterType = "REPLICASET"
```

Comment:
[nit] It's good practice to separate different types into different `const` blocks. This also allows the Go documentation generator to parse comments for such blocks as relevant documentation.

Unofficial example: https://pkg.go.dev/github.com/fluhus/godoc-tricks#Mock_enums